### PR TITLE
learn worlds - username updates

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
- update LearnWorlds sync API to use 'name' instead of 'email' for the LW username
- backfill script for existing users 

## WHY
- so that user emails are not publicized within our LW instance 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/LearnWorlds-Update-Change-process-to-use-Quill-name-instead-of-Quill-email-for-LearnWorlds-Username-85799021ff52407b8b58829355fd58df

### What have you done to QA this feature?
1. Sign up as new LW user; confirm that email is not not displayed as the username within LW
2. run backfill task on staging

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | about to
Self-Review: Have you done an initial self-review of the code below on Github? |

### CODE REVIEW

File: services/QuillLMS/lib/tasks/learn_worlds.rake [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
```
Comment (happythenewsad) [inline services/QuillLMS/lib/tasks/learn_worlds.rake]: This rake task is deletable after a single run

Comment (anathomical) [reply services/QuillLMS/lib/tasks/learn_worlds.rake]: Is it work putting in `lib/tasks/temporary` since it's intended to run once?

**Review by anathomical — APPROVED** [review]
One thought about location for the rake task code, but this looks reasonable to me as-is.

File: services/QuillLMS/app/lib/utils/string.rb [hunk 2]
```diff
@@ -10,6 +10,8 @@ def self.last_name_or_name(first_name_last_name_string)
     last || first
   end
 
+  def self.to_username(str) = str.downcase.strip.gsub(/\s/, '.')
```
Comment (brendanshean) [inline services/QuillLMS/app/lib/utils/string.rb]: I think this method belongs in a LearnWorlds namespace.  app/lib suggests this is a general method that applies to all strings.  I don't think there's a generally accepted to_username method.

**Review by brendanshean — APPROVED** [review]
Looks good to me!

**Pull request merged**